### PR TITLE
Updates to site showcase

### PR DESCRIPTION
**Pull request by kkemple**
Updates made in this PR (related issue #5927):

- [x] refactor site showcase code
- [x] make filters deep-linkable
- [x] make filtering cumulative
- [x] make categories on details links to filtered states

Corresponding issue has been updated to reflect these changes!

### CODE REVIEW

File: www/package.json [hunk 1]
```diff
@@ -77,5 +78,8 @@
     "develop": "gatsby develop",
     "serve": "gatsby serve",
     "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "devDependencies": {
+    "core-js": "^2.5.7"
```
Comment (kkemple) [inline www/package.json]: not sure why this was needed suddenly, saw on issue on preset env about it 🤷‍♀️

File: www/src/components/url-query.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+class URLQuery extends Component {
+  state = {}
+
+  componentDidMount = () => {
+    const {
+      history,
+      location: { search },
+    } = this.props
+
+    this.getDerivedStateFromQuery(search)
```
Comment (kkemple) [inline www/src/components/url-query.js]: too much? :joy:

Comment (sw-yx) [reply www/src/components/url-query.js]: nice... i use a [react-router-state-manager](https://github.com/sw-yx/gatsby/blob/starterShowcase2/www/src/utils/react-router-state-manager.js) utility if you wanna just use that

**Comment by kkemple** [comment]
@sw-yx and @ThatOtherPerson I made some updates to site showcase if you would like to review that would be dope!

**Comment by KyleAMathews** [comment]
I didn't have a chance to look through the code but one change I'd like is that when you click on a facet, it should recalculate all the other facets and only show categories that overlap with the selected facet. E.g. if a site is in two categories, blog and education, then when you click on "blog", "education" should still show and have at least 1 in it.

**Comment by KyleAMathews** [comment]
Link to preview https://deploy-preview-6095--gatsbyjs.netlify.com/showcase/

File: www/src/views/showcase/filtered-showcase.js [hunk 3]
```diff
@@ -0,0 +1,476 @@
...
+import MdCheckboxBlank from "react-icons/lib/md/check-box-outline-blank"
+import MdCheckbox from "react-icons/lib/md/check-box"
+import MdArrowDownward from "react-icons/lib/md/arrow-downward"
+import MdFilterList from "react-icons/lib/md/filter-list"
+import Fuse from "fuse.js"
+import hex2rgba from "hex2rgba"
+import { style } from "glamor"
+
+import ShowcaseList from "./showcase-list"
+import Collapsible from "./collabsible"
```
Comment (tsiq-swyx) [inline www/src/views/showcase/filtered-showcase.js]: typo

**Comment by kkemple** [comment]
hey @sw-yx, thanks for the input!

> there was a lot of moving of files so i can't really tell on a line by line basis what was added or deleted (which i might normally do in a code review)

the only files that are net new are collapsible and showcase list which were classes inside of the showcase page, the only difference between then and now I removed any styles from the declaration that were not used in the new file. Everything seems to be showing line level diffs so if there is something else specifically you were unclear about let me know!

> mobile view
(maybe optional for mvp) filter still not implemented? i am ok with pushing this til after launch
(maybe optional for mvp) detail view: next/previous button?

These aren't completed yet and title hasn't been adjusted yet, those should be in before MVP but I switched focus to v2 for a bit so wanted to get these merged before they fell really far out of sync.

> helmet
details page has weird commas in the title, seems funky
share links
they currently link to the person's page, with no mention of gatsby. i put a mention of gatsby starter in mine and linked to the repo.
misc
this is a broken link: http://localhost:8000/showcase/formidable-com bad formatting?
same here: http://localhost:8000/showcase/thefmg-com

will add to related issue!

**Comment by kkemple** [comment]
@KyleAMathews 

> I didn't have a chance to look through the code but one change I'd like is that when you click on a facet, it should recalculate all the other facets and only show categories that overlap with the selected facet. E.g. if a site is in two categories, blog and education, then when you click on "blog", "education" should still show and have at least 1 in it.

I think that has the potential to cause confusion with the current implementation. We could decouple checkboxes from URL again and try to maintain what is actually set by user and what is set by relation to a selection. What I fear is getting unexpected behavior on 1) selecting filters, i.e. I click on blog to see blogs and portfolio and tech are also selected. 2) I get a showcase link for all portfolio sites but then when page loads I would only get portfolio sites but tech and blog would be selected filters. Curious to here your thoughts!

File: www/src/views/showcase/filtered-showcase.js (hunk 3 above)
Comment (kkemple) [reply www/src/views/showcase/filtered-showcase.js]: oh good catch, is that the filename as well, how is that working then?

**Comment by tsiq-swyx** [comment]
nice work, i will update things on my end

File: www/src/views/showcase/filtered-showcase.js (hunk 3 above)
Comment (tsiq-swyx) [reply www/src/views/showcase/filtered-showcase.js]: yea you matched your typo with another typo haha

**Pull request merged**